### PR TITLE
expo: add option for gesture fingers

### PR DESCRIPTION
**Pull request by leonnelc**
I added a config value for gesture fingers and improved gesture detection by adding some checks so it doesn't get triggered when swiping horizontally or in the opposite direction of gesture_positive, this makes the gesture compatible with workspace swipe to use both gestures with the same number of fingers

### CODE REVIEW

File: hyprexpo/main.cpp [hunk 1]
```diff
@@ -62,39 +62,49 @@ static void hkAddDamageB(void* thisptr, const pixman_region32_t* rg) {
...
-        return;
-
+    static auto* const* PENABLE = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, "plugin:hyprexpo:enable_gesture")->getDataStaticPtr();
+    static auto* const* FINGERS = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, "plugin:hyprexpo:gesture_fingers")->getDataStaticPtr();
     static auto* const* PPOSITIVE = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, "plugin:hyprexpo:gesture_positive")->getDataStaticPtr();
+    auto e = std::any_cast<IPointer::SSwipeUpdateEvent>(param);
 
-    info.cancelled = true;
+    if (!swipeDirection){
+        if (std::abs(e.delta.x) > std::abs(e.delta.y)){
```
Comment (vaxerski) [inline hyprexpo/main.cpp:80]: no {} around short ifs here and further

**Review by vaxerski — COMMENTED** [review]
generally ok but needs a clang-format

File: hyprexpo/main.cpp (hunk 1 above)
Comment (leonnelc) [reply hyprexpo/main.cpp]: done

File: hyprexpo/main.cpp [hunk 2]
```diff
@@ -17,12 +17,12 @@ inline CFunctionHook* g_pAddDamageHookB      = nullptr;
...
-typedef void (*origRenderWorkspace)(void*, CMonitor*, PHLWORKSPACE, timespec*, const CBox&);
-typedef void (*origAddDamageA)(void*, const CBox*);
-typedef void (*origAddDamageB)(void*, const pixman_region32_t*);
-typedef void (*origSwipeBegin)(void*, wlr_pointer_swipe_begin_event*);
-typedef void (*origSwipeEnd)(void*, wlr_pointer_swipe_end_event*);
-typedef void (*origSwipeUpdate)(void*, wlr_pointer_swipe_update_event*);
+typedef void          (*origRenderWorkspace)(void*, CMonitor*, PHLWORKSPACE, timespec*, const CBox&);
+typedef void          (*origAddDamageA)(void*, const CBox*);
+typedef void          (*origAddDamageB)(void*, const pixman_region32_t*);
+typedef void          (*origSwipeBegin)(void*, wlr_pointer_swipe_begin_event*);
```
Comment (vaxerski) [inline hyprexpo/main.cpp:23]: arent these outdated and unused? we can remove the swipe ones right?

Comment (leonnelc) [reply hyprexpo/main.cpp]: yeah, removed.

**Comment by leonnelc** [comment]
also i'm pushing a partial fix for #111, it fixes the "cancel gesture" problem

**Review by vaxerski — APPROVED** [review]
thanks!

**Pull request merged**